wdl-lint/src/rules/redundant_input_assignment.rs

```diff
@@ -0,0 +1,107 @@
+//! A lint rule for flagging redundant input assignments
+
+use std::fmt::Debug;
+
+use wdl_ast::Ast::V1;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Node::NameRef;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use wdl_ast::v1::CallStatement;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the Redundant Input Assignment rule.
+const ID: &str = "RedundantInputAssignment";
+
+/// Create a "Redundant Input Assignment" diagnostic.
+fn redundant_input_assignment(span: Span) -> Diagnostic {
+    Diagnostic::note("redundant input assignments can be shortened")
+        .with_rule(ID)
+        .with_label("redundant input assignments can be shortened", span)
+}
+
+/// Detects a malformed lint directive.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct RedundantInputAssignment(Option<SupportedVersion>);
+
+impl Rule for RedundantInputAssignment {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags redundant input assignments."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Input assignments that are redundant can be shortened. For example, `{ input: a = a }` \
+         can be shortened to `{ input: a }`."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [wdl_ast::SyntaxKind]> {
+        Some(&[
+            wdl_ast::SyntaxKind::VersionStatementNode,
+            wdl_ast::SyntaxKind::WorkflowDefinitionNode,
+            wdl_ast::SyntaxKind::CallStatementNode,
+        ])
+    }
+}
+
+impl Visitor for RedundantInputAssignment {
+    type State = Diagnostics;
+
+    fn document(
+        &mut self,
+        _: &mut Self::State,
+        reason: VisitReason,
+        _: &Document,
+        version: SupportedVersion,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // Reset the visitor upon document entry
+        *self = Self(Some(version));
+    }
+
+    fn call_statement(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        stmt: &CallStatement,
+    ) {
+        if let SupportedVersion::V1(minor_version) = self.0.expect("version should exist here") {
+            if minor_version < V1::One {
```

Comment:
@a-frantz  This line is giving me a problem, where `V1::One` is erroring, returning: `failed to resolve: `V1` is a variant, not a module`. I tried to mirror the `missing_requirements.rs` implementation so am confused (what's new!)

Reply:
```suggestion
            if minor_version < wdl_ast::version::V1::One {
```
And then remove from the top of the file `use wdl_ast::Ast::V1;`. You imported the wrong thing (although they are named the same, so that's confusing 😬 )